blog/content/post/2019-07-02-qubes-os-and-3mdeb-minisummit.md

```diff
@@ -0,0 +1,192 @@
+  firmware level. It's good to know that hardware Qubes OS operate on
+  reasonable quality firmware.
+
+# Xen and Qubes OS status
+
+There was no formal presentation about that topic, but discussion drifted to
+that topic, so it is definitely worth to mention. Of course we are not
+hard-core Xen or Qubes OS developers, so some of those information may be not
+complete:
+
+* There are some issues with PCI pass-through. Qubes OS would like to use that
+  feature more, but because of complex PCI architecture and design assumptions
+  these days it is not so each and Xen should improve need improvements in that
+  area
+* There is some problem PCI hot-plug since all connected devices automatically
+  appear in dom0, which can cause some security issues if malicious device
+  would be connected
+* USB Type C is another problem since host controller is visible only after
+  connecting device, so OS doesn't have control over it before
+* It seems that correct support for IOMMU can mitigate most of above problems
+* There are very interesting improvement in new microarchitectures like
+  Denverton, especially VT-c, but Xen still doesn't have support for that
+* There are also know issues in GPU support e.g. Nvidia requires dedicated,
+  more expensive hardware, if it would be used in virtualized environment
+* Because of above problems and other reasons Qubes OS community consider KVM
+  support
+
+We also discussed how 3mdeb can involve in improving Qubes OS and Xen and of
+course community activity would be best approach. Of course as for-profit
+organization we have to align that with some business activity. We plan to work
+on that in 2019. As first step towards that is this blog and following ideas
+for contribution.
+
+# Qubes OS build system
+
+According to [documentation](https://www.qubes-os.org/doc/qubes-builder/) Qubes
+OS should be build using rpm-based distro like Fedora. According to Marek build
+should also work on Debian without big issues. Definitely it is something to
+check and maybe improve documentation.
+
+But moving forward discussion about Qubes OS build system, as [Yocto Participants](https://www.qubes-os.org/doc/qubes-builder/),
+we advertised OE/Yocto as build system for dom0 in Qubes OS. During that
+discussion various problems pop up:
+
+* OE/Yocto not verify hashes or does it in doubtful way.
+* We can't freeze dom0, we need flexibility in extending it additional software
+  through package manager - as Marek mentioned this statement is valid until
+  GUI is in dom0. Goal is definitely to get rid of GUI from dom0.
+* We should treat big distros like Debian and Fedora as role models in package
+  management, they already use lots of good practice that are not followed by
+  OE/Yocto, because of that 3mdeb should understand better how Debian/Fedora
+  does package management and help reflect best practice in OE/Yocto to improve
+  probability of using it as dom0 build system.
```

Comment:
Very good analysis about package managers and updates is here: https://theupdateframework.github.io/
Especially this paper: https://theupdateframework.github.io/papers/package-management-security-tr08-02.pdf?raw=true